sql/core/src/test/scala/org/apache/spark/sql/execution/QueryExecutionSuite.scala

```diff
@@ -160,7 +160,9 @@ class QueryExecutionSuite extends SharedSparkSession {
 
     // Throw an AnalysisException - this should be captured.
     spark.experimental.extraStrategies = Seq[SparkStrategy](
-      (_: LogicalPlan) => throw new AnalysisException("_LEGACY_ERROR_TEMP_3078", Map.empty))
+      (_: LogicalPlan) => throw new AnalysisException(
```

Comment:
The UT can actually handle any type of `AnalysisException`. I have chosen an error class (`UNSUPPORTED_DATASOURCE_FOR_DIRECT_QUERY`) that is closer to semantics to replace it.